runtime/src/accounts_db.rs

```diff
@@ -1109,11 +1143,24 @@ impl AccountsDB {
     fn hash_accounts(&self, slot_id: Slot, accounts: &[(&Pubkey, &Account)]) -> Vec<Hash> {
         let mut hash_state = BankHash::default();
         let mut had_account = false;
+        let mut stats = BankHashStats::default();
         let hashes: Vec<_> = accounts
             .iter()
             .map(|(pubkey, account)| {
                 if !sysvar::check_id(&account.owner) {
                     let hash = BankHash::from_hash(&account.hash);
+                    if hash.is_default() {
```

Comment:
nits: This hunk can also be extracted to `impl BankHashStats` and used like this `stats.update(account)`, which reads nice imo.